src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -225,7 +226,7 @@ public void setRate(final double throughPutMbPerSec)
     private boolean hasEnoughCompactionsRunning(ColumnFamilyStore cfs)
     {
         int count = compactingCF.count(cfs);
-        if (count > 0 && executor.getActiveCount() >= executor.getMaximumPoolSize())
+        if (count > 0 && executor.exactRunningTasks.get() >= executor.getMaximumPoolSize())
```

Comment:
This is still possibly undercounting (don't we actually need the number of scheduled rather than running tasks?), but that's fully ok.

Let's add a comment (maybe in JavaDoc) that this may return a false negative, because scheduling too many next-compaction-checks is not a real problem. On the other hand, returning a false positive can cause us to unnecessarily postpone compactions that we can safely start.

Reply:
Actually, `getActiveCount` returns the number of running tasks rather than scheduled, so the meaning of this is not changed - I only changed the way how we obtain the count of active tasks (in the second commit I've moved counting to `beforeExecution` and `afterExecution` methods)